ansible/playbook.yaml

```
@@ -0,0 +1,69 @@
+- name: update web servers
+  hosts: webservers
+  remote_user: ec2-user
+
+  vars_prompt:
+    - name: trello_api_key
+      private: false
+      prompt: What is the Trello API key
+
+    - name: trello_api_token
+      prompt: What is the Trello API token
+
+  tasks:
+  - name: Install the lastest version of Git
+    become: yes
+    ansible.builtin.yum:
+      name: git
+      state: latest
+
+  - name: Install the lastest version of Python
+    become: yes
+    ansible.builtin.yum:
+      name: python311
+      state: latest
+
+  - name: Execute the command in remote shell to install Poetry
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args:
+      creates: /home/ec2-user/.local/bin/poetry
+
+  - name: Create Application Folder
+    become: yes
+    ansible.builtin.shell: mkdir /opt/todoapp
+    args:
+      creates: /opt/todoapp
+
+  - name: Change App folder owner to ec2-user
+    become: yes
+    ansible.builtin.shell: chown ec2-user /opt/todoapp
```

This works well, but worth noting that [the inbuilt file module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html) can handle these two steps together, and potentially accounts for side-effects that we might have missed (e.g. I don't know, but does this correctly identify when the permissions change, vs when they are already set?)

```suggestion
  - name: Create Application Folder
    become: yes
    ansible.builtin.file:
      path: /opt/todoapp
      state: directory
      owner: ec2-user
```